wrappers/src/shared_realm_cs.cpp

```diff
@@ -369,6 +374,41 @@ REALM_EXPORT uint64_t shared_realm_get_schema_version(SharedRealm& realm, Native
     });
 }
 
+REALM_EXPORT uint32_t shared_realm_begin_transaction_async(SharedRealm& realm, void* tcs_ptr, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        return realm->async_begin_transaction([tcs_ptr, ex]()
+        {
+            s_handle_write_commit_async(tcs_ptr, ex);
+        }, true);
+    });
+}
+
+REALM_EXPORT uint32_t shared_realm_commit_transaction_async(SharedRealm& realm, void* tcs_ptr, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        return realm->async_commit_transaction([tcs_ptr](std::exception_ptr err) {
+            NativeException::Marshallable nativeEx { RealmErrorType::NoError };
```

Comment:
Isn't `NoError` the default here?

Reply:
If you're referring to the fact that `handle_error` sets the type to `NoError`. That's correct. If you are just asking in general, then no because `NoError` is -1. (You can see it [here](https://github.com/realm/realm-dotnet/blob/67e05668a650b2aa7e5e262fc652d9e29aac478c/wrappers/src/realm_error_type.hpp#L27))
Either way, are you suggesting that being as implicit as possible is what we should strive for? That doesn't sound as a good practice as it can be one more source of bugs, plus while reading it'd require browsing around (as I had to do) to see the default value.
Have I misunderstood something?

Reply:
I would have preferred that a default-constructed `NativeException::Marshallable` sets the exception type to no error, but since that's not the case, then the code as it is right now is just fine.